.github/files/update-jetpack-staging-sites.sh

```diff
@@ -0,0 +1,143 @@
+    "url": "https://jetpackedgeprivate.wpcomstaging.com/",
+    "note": "private site",
+    "ssh_string": "jetpackedgeprivate.wordpress.com@sftp.wp.com",
+    "blog_id": "215380534"
+  },
+  "jetpackedgewpbeta.wpcomstaging.com": {
+    "url": "https://jetpackedgewpbeta.wpcomstaging.com/",
+    "note": "latest wp beta",
+    "ssh_string": "jetpackedgewpbeta.wordpress.com@sftp.wp.com",
+    "blog_id": "215380197"
+  },
+  "jetpackedgewpprevious.wpcomstaging.com": {
+    "url": "https://jetpackedgewpprevious.wpcomstaging.com/",
+    "note": "previous wp version",
+    "ssh_string": "jetpackedgewpprevious.wordpress.com@sftp.wp.com",
+    "blog_id": "215380213"
+  }
+}'
+
+####################################################
+## Fetch plugin data from the Jetpack Beta Builder.
+####################################################
+
+# Fetch the latest data from the Jetpack Beta Builder for each plugin.
+for PLUGIN in "${PLUGINS[@]}"; do
+  echo "Fetching latest $PLUGIN data from the Jetpack Beta Builder..."
+  if ! RESPONSE=$(curl -s https://betadownload.jetpack.me/$PLUGIN-branches.json); then
+    echo "Error: unable to fetch data from Jetpack Beta Builder for $PLUGIN."
+    exit 1
+  fi
+
+  DOWNLOAD_URL=$(echo "$RESPONSE" | jq -r ".master.download_url")
+  PLUGIN_VERSION=$(echo "$RESPONSE" | jq -r ".master.version")
+
+  if [[ -z "$DOWNLOAD_URL" || -z "$PLUGIN_VERSION" ]]; then
+    echo "Error: unable to extract data from Jetpack Beta Builder response for $PLUGIN."
+    exit 1
+  else
+    echo "Returned version number for $PLUGIN: $PLUGIN_VERSION"
+    PLUGIN_VERSIONS[$PLUGIN]="$PLUGIN_VERSION"
+    PLUGIN_DOWNLOAD_URLS[$PLUGIN]="$DOWNLOAD_URL"
+  fi
+done
+
+####################################################
+## Download and unpack the plugin .zip files.
+####################################################
+
+for PLUGIN in "${PLUGINS[@]}"; do
+  echo "Attempting to download $PLUGIN .zip file..."
+  if ! curl -f -o "$TMP_DIR/$PLUGIN-dev.zip" "${PLUGIN_DOWNLOAD_URLS[$PLUGIN]}"; then
+    echo "Download of $PLUGIN .zip failed, exiting."
+    exit 1
+  else
+    echo "Download of $PLUGIN .zip completed."
+  fi
+
+  echo "Unpacking .zip file to: $TMP_DIR/$PLUGIN"
+  if ! unzip -q "$TMP_DIR/$PLUGIN-dev.zip" -d "$TMP_DIR"; then
+    echo "Unpacking of the .zip failed, exiting."
+    exit 1
+  else
+    echo "Unpacking of $PLUGIN completed successfully."
+    # Add a version.txt file for easy curl retrieval on the MC staging-updater page.
+    echo "${PLUGIN_VERSIONS[$PLUGIN]}" > "$TMP_DIR/$PLUGIN-dev/version.txt"
+  fi
+done
+
+####################################################
+## Sync the new Jetpack files.
+####################################################
+
+# Write the SSH key to temp file for use in the rsync command.
+echo "$UPDATEJETPACKSTAGING_SSH_KEY" > "$TMP_DIR/ssh_key"
+chmod 0600 "$TMP_DIR/ssh_key"
+
+# Sync new plugin files to the Atomic test sites.
+EXIT_CODE=0
+for key in $(echo "${SITES}" | jq -r 'keys[]'); do
```

Comment:
Seems slightly cleaner to do like this.

```suggestion
for key in $(jq -r 'keys[]' <<<"$SITES"); do
```
Same below.